src/textual/widgets/_text_area.py

```diff
@@ -2199,6 +2202,33 @@ def action_delete_line(self) -> None:
         if deletion is not None:
             self.move_cursor_relative(columns=end_column, record_width=False)
 
+    def action_cut(self) -> None:
+        """Cut text (remove and copy to clipboard)."""
+        if self.read_only:
+            return
+        start, end = self.selection
+        if start == end:
+            return
+        copy_text = self.get_text_range(start, end)
```

Comment:
I think the "delete_via_keyboard" method returns the text that was deleted as part of the ExitResult, so you shouldn't need to get the range again here.

See "_on_paste"